modules/lang-painless/src/test/java/org/elasticsearch/painless/ElvisTests.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.painless;
+
+import static java.util.Collections.singletonMap;
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.lessThan;
+
+/**
+ * Tests for the Elvis operator ({@code ?:}).
+ */
+public class ElvisTests extends ScriptTestCase {
+    public void testBasics() {
```

Comment:
@nik9000 What do you think about this proposal for now?

What if for the null safe operator (?.), if the guarded type is a primitive we disallow it, and then for the elvis operator (?:) if the lhs is a primitive we disallow that too.  I honestly don't think anyone will notice because currently there is no way to get a primitive out of a field from a non-static type (nothing exists in the whitelist afaik), and I would argue in the case where you want a primitive out of a call, it doesn't make sense to have the receiver be a boxed type.  You would still have to check to see if the receiver was null afterwards anyway.  For most cases the type will be def and the auto-boxing will just happen anyway.  Both of these operators can be very useful for def types without needing to have them do magic for primitives.  I would hate to not have them because of boxing issues when it's improbable that users would run into them.